pyinfra/operations/apt.py

```diff
@@ -21,21 +27,18 @@
 APT_UPDATE_FILENAME = "/var/lib/apt/periodic/update-success-stamp"
 
 
-def noninteractive_apt(command: str, force=False):
-    args = ["DEBIAN_FRONTEND=noninteractive apt-get -y"]
-
-    if force:
-        args.append("--force-yes")
+def _simulate_then_perform(command: str):
+    changes = host.get_fact(SimulateOperationWillChange, command)
```

Comment:
Regarding your question that you asked on matrix, I think you should do:
 changes = host.get_fact(SimulateOperationWillChange, command=command)

Reply:
My problem is, that the `get_fact` method of the `FakeHost` of the framework does not support positional arguments.
But there are already other facts that use positional arguments, so this seems to be an oversight :thinking:

Reply:
Oh, that's a good point, I only though about the definiton of the `command` method using positional arguments, e.g. for `FindInFile`:

```python
def command(self, path, pattern, interpolate_variables=False):
```

But even that is called using a keyword argument:

```python
pyinfra/operations/server.py
19:from pyinfra.facts.files import Directory, FindInFile, Link
1193:        FindInFile, path=locales_definitions_file, pattern=rf"^.*{locale}[[:space:]]\+.*$"
```

So, maybe that's an oversight or some quirk that should be documented ?

In the meantime I added some rudimentary support (or rather, ignorance of) positional arguments for `get_fact`

Reply:
And the real `get_fact` method does support passing in positional arguments, but then apparently does some wild magic to "merge" positional and keyword arguments: 

https://github.com/pyinfra-dev/pyinfra/blob/3.x/pyinfra/api/facts.py#L121-L143

Reply:
You're right, it is possible, just not used :)
You're fix in tests/utils.py seems justified :+1:

Reply:
> Agreed this is a good change! The facts API is... old and messy, unfortunately hard to fix it without breaking it.

I mean, if you ever do decide to change it (in a probably breaking way), one of the things that's been bothering me is, that while you usually use operations in a "namespace preserving way" i.e. `files.put` you don't do the same with Facts, i.e. `FileExists` ...